### PR TITLE
feat: update earn table

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Reworked earn table to include more data
- Added an asset teaser popover
- Updated the latest version of framer-motion to fix a bug in the older version

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.

## Risk

Outline the scope of your changes and the risk associated with them. You must use your discretion as an engineer to determine the potential impact of your changes.

E.g. an upgrade to `hdwallet` or core state management would be considered higher risk, and might require a full regression test. UI or isolated view changes, or something behind a feature flag may have near zero risk. Small bug fixes might require testing isolated to the specific fix.

## Testing

If your testing steps are technical, outline steps for an engineer to verify.

If your testing steps are functional, please provide a full guide with any features requiring special attention for operations to test your branch in the preview environment.

## Screenshots (if applicable)
Screen Shot 2022-02-16 at 10 56 24 AM
Screen Shot 2022-02-16 at 10 56 36 AM
Screen Shot 2022-02-16 at 10 56 48 AM

### CODE REVIEW

File: src/components/StakingVaults/EarnOpportunityRow.tsx [hunk 1]
```diff
@@ -95,67 +131,93 @@ export const EarnOpportunityRow = ({
     yearn
   ])
 
+  const debouncedHandleMouseEnter = debounce(() => setShowPopover(true), 100)
+  const handlOnMouseLeave = () => {
+    debouncedHandleMouseEnter.cancel()
+  }
```
Comment (0xdef1cafe) [inline src/components/StakingVaults/EarnOpportunityRow.tsx:137]:
```suggestion
  const handleOnMouseLeave = debouncedHandleMouseEnter.cancel
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
visually looks good but code changes required

File: src/components/StakingVaults/EarnOpportunities.tsx [hunk 2]
```diff
@@ -38,25 +39,41 @@ export const EarnOpportunities = ({ assetId: caip19 }: EarnOpportunitiesProps) =
...
-        <Stack spacing={2} mt={2} mx={-4}>
-          {vaults.map(vault => (
-            <EarnOpportunityRow {...vault} key={`${vault.vaultAddress}`} isLoaded={!!vault} />
-          ))}
-        </Stack>
+      <Card.Body pt={0} px={2}>
+        <Table variant='clickable'>
+          <EarnTableHeader />
+          <Tbody>
+            {vaults.map((vault, index) => (
```
[the remark this answers is not in the document]
Comment (reallybeard) [reply src/components/StakingVaults/EarnOpportunities.tsx:67]: The second part I think is a little out of scope of this, the yearn vaults need to be re-worked to allow this to happen.

File: src/components/StakingVaults/AssetTeaser.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+export const AssetTeaser = ({ assetId }: { assetId: CAIP19 }) => {
+  const asset = useAppSelector(state => selectAssetByCAIP19(state, assetId))
+  const { description, icon, name } = asset || {}
+  const { isLoading } = useGetAssetDescriptionQuery(assetId, { skip: !!description })
+  const url = useMemo(() => {
+    if (!assetId) return ''
+    const { chain, tokenId } = caip19.fromCAIP19(assetId)
+    let baseUrl = `/assets/${chain}`
+    if (tokenId) baseUrl = baseUrl + `/${tokenId}`
```
[the remark this answers is not in the document]
Comment (reallybeard) [reply src/components/StakingVaults/AssetTeaser.tsx:31]: This follows the pattern we established with the AccountRow component.

File: package.json [hunk 4]
```diff
@@ -50,7 +50,7 @@
     "dayjs": "^1.10.6",
     "dompurify": "^2.3.2",
     "envalid": "^7.2.1",
-    "framer-motion": "^4",
+    "framer-motion": "4.1.16",
```
Comment (0xdef1cafe) [inline package.json:53]:
```suggestion
    "framer-motion": "^4.1.16",
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Review by DaoDev44 — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**